### PR TITLE
localize when `wp core language activate`

### LINKED ISSUE #3731: Improve `wp core language activate`
We need to set default timezone and date format.

There is a time lag from en_US version to localized version, so we can't run `wp core install --locale=ja` today.

Following works fine, but timezone and date format won't be set.

```
$ wp core install
$ wp core language install ja --activate
```

Default timezone and format is in `po/mo` files.

* https://translate.wordpress.org/projects/wp/dev/admin/ja/default?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=40444&filters%5Btranslation_id%5D=1653450
* https://translate.wordpress.org/projects/wp/dev/admin/ja/default?filters%5Bstatus%5D=either&filters%5Boriginal_id%5D=216367&filters%5Btranslation_id%5D=10294358

I will write a patch.

### PR DESCRIPTION
**Pull request by miya0001**
Change timezone string and date_format when language is activated.

Fixes #3731

### CODE REVIEW

**Comment by danielbachhuber** [comment]
> Change timezone string and date_format when language is activated.

Does WordPress normally do this in the admin?

**Comment by miya0001** [comment]
No, it doesn't. It will be executed at the time of a new installation with localized version.
But we can't use localized version when it hasn't been released.

If you are worried about this, I think it can add some command line options about it.

**Comment by danielbachhuber** [comment]
> No, it doesn't. It will be executed at the time of a new installation with localized version.

Ok. I don't think it makes sense to deviate from core's behavior.

While helpful to some, changing the timezone and date format may be unexpected to others.

> But we can't use localized version when it hasn't been released.

How long does it typically take to release the localized version?

**Comment by miya0001** [comment]
> How long does it typically take to release the localized version?

Maybe 2 days or so, but some country is maybe later.

**Comment by danielbachhuber** [comment]
Ok. What do you think about making this a `--set-date-time` flag?

**Comment by miya0001** [comment]
Nice! I am very happy if this problem will be solved.

**Comment by miya0001** [comment]
I've added `--set-date-time` option and update tests.

File: features/core-language.feature [hunk 1]
```diff
@@ -117,15 +117,55 @@ Feature: Manage translation files for a WordPress install
...
+      """
+      j M Y
+      """
+
+    When I run `wp option get timezone_string`
+    Then STDOUT should contain:
+      """
+      Europe/Istanbul
+      """
+
```
Comment (danielbachhuber) [inline features/core-language.feature]: Don't we need a test that verifies `date_format` and `timezone_string` aren't set when `--set-date-time` isn't passed?

File: php/WP_CLI/CommandWithTranslation.php [hunk 2]
```diff
@@ -147,7 +150,8 @@ public function install( $args, $assoc_args ) {
 			\WP_CLI::success( "Language installed." );
 
 			if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'activate' ) ) {
-				$this->activate( array( $language_code ), array() );
+				$set_date_time = \WP_CLI\Utils\get_flag_value( $assoc_args, 'set-date-time' );
+				$this->activate( array( $language_code ), array( 'set-date-time' => $set_date_time ) );
```
Comment (danielbachhuber) [inline php/WP_CLI/CommandWithTranslation.php]: Let's only pass `array( 'set-date-time' => $set_date_time )` when it's a truthy value.

**Comment by miya0001** [comment]
Thanks for the review. I updated them.

**Comment by miya0001** [comment]
@danielbachhuber Are there any work to solve error on Travis?

**Comment by miya0001** [comment]
@danielbachhuber 

I found some problems in this PR, and it does not solve our problem.
I want to rethink and close it.

Thanks!

**Comment by danielbachhuber** [comment]
> I found some problems in this PR, and it does not solve our problem.
> I want to rethink and close it.

Ok

**Pull request closed without merging**